src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -576,11 +576,22 @@ public function whereBelongsTo($related, $relationshipName = null, $boolean = 'a
             throw RelationNotFoundException::make($this->model, $relationshipName, BelongsTo::class);
         }
 
-        $this->whereIn(
-            $relationship->getQualifiedForeignKeyName(),
-            $relatedCollection->pluck($relationship->getOwnerKeyName())->toArray(),
-            $boolean,
-        );
+        if ($relationship instanceof MorphTo) {
+            $this->where(fn ($query) => (
+                $relatedCollection->each(fn ($model) =>
+                    $query->orWhere(fn ($innerQuery) =>
+                        $innerQuery->where($relationship->getQualifiedMorphTypeColumn(), \get_class($model))
```

Comment:
Wouldn't think exclude relation morph map?

Reply:
I just realized that there's a [`whereMorphedTo`](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php#L462) although it's not mentioned anywhere in the docs and the signature is different than whereBelongsTo although it looked like it could be used to accomplish the same thing.

Should I rework this part to use `whereMorphedTo` (seeing as it would've handled relationship morph map nicely) instead?

Reply:
Might be better to update the documentation instead of new code, don't you think?

Reply:
Alright, I'll help with updating the documentation. You can close this PR in the meantime.

I'm thinking of adding it to Eloquent Relationships at the "Querying Morph to Relationship" section ((git)[https://github.com/laravel/docs/blob/11.x/eloquent-relationships.md#querying-morph-to-relationships]/(docs)[https://laravel.com/docs/11.x/eloquent-relationships#querying-morph-to-relationships]), but I'm open to suggestion if there's a better place to put it.